### ISSUE TITLE
[QUESTION] Download Low Fidelity Version of Video? Mp2t Files?

### ISSUE THREAD

**Issue by SiteDesignUSA** [nishad/udemy-dl-windows#40]
Is there a switch for selecting low fi version of video if available?

Does this handle the new mp2t files that Udemy is implementing?

**Comment by mukhlesur-millon** [comment]
could you please comment here the course link which has Mp2t files.

**Comment by nishad** [comment]
`mp2t` or `.ts` is a **transport stream** with MPEG-2 video under RFC-3555, not a file type. With the current implementation of `udemy-dl`, it is hard to include such streams.

About selecting video quality, I'll consider if more users are demanding for it. But these changes should be implemented first in https://github.com/nishad/udemy-dl.

**Comment by SiteDesignUSA** [comment]
Last week (21st Jan) all of a sudden there was no choice for video quality playback on a couple of the courses that I bought. I used to have a choice to go down to 360 playback because my DSL is pitiful. But that changed. All there was to choose was "automatic". I checked out (Inspector) what was being downloaded and they were chunks (mp4) directed by these mp2t files and my browsers were attempting to download all stream sizes from highest to lowest at the same time. Needless to say, that slowed things a bit.

That's when I found your tool. (By the way thanks!) Anyway it seems Udemy had put something into effect and the lead left for the weekend leaving many (like me) without video. They fixed Monday and all is back to normal, but thanks for the tool. I was able to finish my course over the weekend because of you.

Like I told Udemy, we all don't have 12mbs connections. Your tool really helped me.

**Comment by SiteDesignUSA** [comment]
By the way, tonight I tried to download course (which I bought) and received 

https://github.com/nishad/udemy-dl-windows/issues/40

and

Traceback (most recent call last):
  File "<string>", line 374, in <module>
  File "<string>", line 370, in main
  File "<string>", line 294, in udemy_dl
AttributeError: 'NoneType' object has no attribute 'encode'

Errors. I read the other threads and saw the comments about having access to them and I did purchase them.

I'll post over there too.
